container-storage-setup.sh

```diff
@@ -694,6 +747,47 @@ canonicalize_block_devs() {
   echo "$devs_abs"
 }
 
+# This is used in config creation mode
+canonicalize_block_devs_generic() {
+  local devs=$1 dev
+  local devs_resolved dev_resolved
```

Comment:
I don't have a great suggestion, but would ask you to consider renaming devs_resolved or dev_resolved.  They are so similar in naming that I could see them becoming confused or mistyped in later maintenance.  Perhaps simply devs_resolved to resolved_devs just for the sake of differentiation?